debezium-core/src/main/java/io/debezium/config/Configuration.java

```diff
@@ -2125,18 +2125,12 @@ default String getFallbackStringProperty(Field newProperty, Field oldProperty) {
      * @return the evaluated value
      */
     static String getFallbackStringProperty(Configuration config, Field newProperty, Field oldProperty) {
-        if (null != config.getString(oldProperty.name()) && null != config.getString(newProperty.name())) { // both are set
+        if (null != config.getString(oldProperty.name()) || null != config.getString(newProperty.name())) { // either one is set
             CONFIGURATION_LOGGER.warn("Provided configuration has deprecated property \"" + oldProperty.name()
                     + "\" and new property \"" + newProperty.name() + "\" set. Using value from \"" + newProperty.name() + "\"!");
         }
         return config.getString(
                 newProperty,
-                () -> {
-                    CONFIGURATION_LOGGER.warn("Using configuration property \"" + oldProperty.name()
-                            + "\" is deprecated and will be removed in future versions. Please use \"" + newProperty.name()
-                            + "\" instead.");
-                    return config.getString(oldProperty);
-                });
+                () -> config.getString(oldProperty));
```

Comment:
I don't think this yields the behavior I would have expected.  I would expect:

* If both are present, we log a warning about prioritizing the new value over the old.
* If the old is present, we log a warning about the property being deprecated.
* If only the new is present, we log nothing.

```suggestion
        if (null != config.getString(oldProperty.name())) {
            if (null != config.getString(newProperty.name())) { // both are set
                CONFIGURATION_LOGGER.warn("Provided configuration has deprecated property \"" + oldProperty.name()
                        + "\" and new property \"" + newProperty.name() + "\" set. Using value from \"" + newProperty.name() + "\"!");
            }  else { // old value is set
                CONFIGURATION_LOGGER.warn("Using configuration property \"" + oldProperty.name()
                        + "\" is deprecated and will be removed in future versions. Please use \"" + newProperty.name()
                        + "\" instead.");            
            }
        }

        return config.getString(newProperty, () -> { return config.getString(oldProperty));
```

Reply:
Thanks @Naros, added the suggestions.